data-prepper-plugins/s3-sink/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/JsonRecordsGenerator.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.time.Instant;
+import java.util.Map;
+import java.util.UUID;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.plugins.sink.codec.Codec;
+import org.opensearch.dataprepper.plugins.sink.codec.JsonCodec;
+import com.fasterxml.jackson.core.JsonEncoding;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+
+class JsonRecordsGenerator implements RecordsGenerator {

```

Comment:
You don't need this class. The S3 sink is responsible for generating the JSON - not the test.

You may want to have a `JsonRecordsReader` which can read an actual object from S3.

Reply:
Removed two unused files related to `RecordsGenerator` and `JsonRecordsGenerator`.